interface/ceed-basis.c

```diff
@@ -1102,6 +1102,84 @@ int CeedBasisGetGrad(CeedBasis basis, CeedScalar* *grad) {
   return 0;
 }
 
+/**
+  @brief Get value in CeedEvalMode matrix of a CeedBasis
+
+  @param basis       CeedBasis
+  @param[out] value  Variable to store value
+  @param[in] emode   CeedEvalMode to retrieve value
+  @param[in] node    Node (column) to retrieve value
+  @param[in] qpt     Quadrature point (row) to retrieve value
+  @param[in] dim     Dimension to retrieve value for, for CEED_EVAL_GRAD
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Advanced
+**/
+int CeedBasisGetValue(CeedBasis basis, CeedScalar *value, CeedEvalMode emode,
+                      CeedInt node, CeedInt qpt, CeedInt dim) {
```

Comment:
Output arg order fixed.
I intentionally stayed away from building the entire basis matrix in case its large. I don't have a good sense of the performance tradeoffs for building a potentially rather large matrix vs some possibly recomputing some values.

Reply:
You can't get fast code when using scalars and mixing compute-on-the-fly into a tiling scheme would be a mess. I would just build the whole damn matrix (it's faster to compute the entries that way anyway, and it can be reused across all elements). It'll be useful (for performance testing purposes and some modest order problems) to be able to assemble the true element matrices, which PETSc can assemble into a sparse matrix. If you get the matrix, that can be done using GEMM.

Reply:
Let's stick with the current interface. My only other comments is that we usually index `(row,col)` versus `(col,row)`, so I think the standard ordering would be more clear.

Reply:
Switched